modules/workers/instancepools.tf

```diff
@@ -61,3 +61,63 @@ resource "oci_core_instance_pool" "workers" {
     }
   }
 }
+
+resource "oci_core_instance_pool" "autoscaled_workers" {
+  # Create an OCI Instance Pool resource for each enabled entry of the worker_pools map with that mode.
+  for_each                  = { for key, value in local.enabled_instance_pools: key => value if lookup(value, "ignore_size", false) == true }
+  compartment_id            = each.value.compartment_id
+  display_name              = each.key
+  size                      = each.value.size
+  instance_configuration_id = oci_core_instance_configuration.workers[each.key].id
+  defined_tags              = each.value.defined_tags
+  freeform_tags             = each.value.freeform_tags
+
+  dynamic "placement_configurations" {
+    for_each = each.value.availability_domains
+    iterator = ad
+
+    content {
+      availability_domain = ad.value
+      primary_subnet_id   = each.value.subnet_id
+
+      # Value(s) specified on pool, or null to select automatically
+      fault_domains = try(each.value.placement_fds, null)
+
+      dynamic "secondary_vnic_subnets" {
+        for_each = lookup(each.value, "secondary_vnics", {})
+        iterator = vnic
+        content {
+          display_name = vnic.key
+          subnet_id    = lookup(vnic.value, "subnet_id", each.value.subnet_id)
+        }
+      }
+    }
+  }
+
+  lifecycle {
+    ignore_changes = [
+      display_name, defined_tags, freeform_tags,
+      placement_configurations, size
+    ]
+
+    precondition {
+      condition     = coalesce(each.value.image_id, "none") != "none"
+      error_message = <<-EOT
+      Missing image_id; check provided value if image_type is 'custom', or image_os/image_os_version if image_type is 'oke' or 'platform'.
+        pool: ${each.key}
+        image_type: ${coalesce(each.value.image_type, "none")}
+        image_id: ${coalesce(each.value.image_id, "none")}
+      EOT
+    }
+
+    precondition {
+      condition     = var.cni_type == "flannel"
+      error_message = "Instance Pools require a cluster with `cni_type = flannel`."
+    }
+
+    precondition {
+      condition     = each.value.autoscale == false
```

Comment:
We should remove this precondition - user could be configuring CA on their own/through add-on and still need the ability to `ignore_size`

Reply:
@devoncrouse If users want to install/configure CA on their own, then they should just set the `ignore_size` variable to `true` and `autoscale` here to `false`. The value of `autoscale` here only impacts the generated autoscaler template to let it know which nodepool/instancepool it should manage. Not whether the nodepools/instancepools themselves.

Am I missing something?

Reply:
@hyder You're right, that should be fine - thanks